tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -176,75 +176,122 @@ def test_login_node_security_groups(
+    )
 
     if assign_additional_security_groups:
         security_groups = describe_cluster_security_groups(cluster.name, region)
-        load_balancer_managed_security_group = None
-        login_node_managed_security_group = None
-        for security_group in security_groups:
-            if "pool1LoadBalancerSecurityGroup" in security_group.get("GroupName"):
-                load_balancer_managed_security_group = security_group
-            if "pool1LoginNodesSecurityGroup" in security_group.get("GroupName"):
-                login_node_managed_security_group = security_group
-
-        logging.info("Asserting both the login node and load balancer have managed security group")
-
-        assert_that(
-            load_balancer_managed_security_group is not None and login_node_managed_security_group is not None
-        ).is_true()
-
-        assert_that(
-            any(
-                login_node_managed_security_group.get("GroupId") == security_group["GroupId"]
-                for security_group in instances[0].get("SecurityGroups")
-            )
-        ).is_true()
-
-        assert_that(
-            load_balancer_managed_security_group.get("GroupId") in load_balancers[0].get("SecurityGroups")
-        ).is_true()
-
-        logging.info("Asserting the managed security groups of the load balancer and login node share SSH restriction")
-        load_balancer_managed_rules = ec2_client.describe_security_group_rules(
-            Filters=[{"Name": "group-id", "Values": [load_balancer_managed_security_group.get("GroupId")]}]
-        )["SecurityGroupRules"]
-        # Check login node SG
-        check_node_security_group(region, cluster, 22, ssh_from, login_pool_name="pool1")
-        # Check load balancer SG
-        assert_that(
-            any(ssh_from == rule.get("CidrIpv4") and rule.get("FromPort") == 22 for rule in load_balancer_managed_rules)
-        ).is_true()
+        logging.info(f"Security groups: {security_groups}")
 
-        logging.info(
-            "Asserting that load balancer managed security group is referenced in an inbound rule "
-            "of the login node managed security group"
+        # get the managed security groups of the pool's load balancer and nodes
+        pool_1_load_balancer_managed_sg, pool_1_node_managed_sg = _get_pool_managed_security_groups(
+            "pool1", security_groups
         )
-        for ip_permissions in login_node_managed_security_group.get("IpPermissions"):
-            if ip_permissions.get("UserIdGroupPairs"):
-                assert_that(
-                    any(
-                        load_balancer_managed_security_group.get("GroupId") == id_group_pairs.get("GroupId")
-                        for id_group_pairs in ip_permissions.get("UserIdGroupPairs")
-                    )
-                ).is_true()
+        pool_2_load_balancer_managed_sg, pool_2_node_managed_sg = _get_pool_managed_security_groups(
+            "pool2", security_groups
+        )
```

Comment:
you could use a for loop that goes through the 2 pools so that you don't have to write each assert twice